Replicator/IncomingRev.cc

```diff
@@ -273,16 +347,31 @@ namespace litecore { namespace repl {
         // Call the custom validation function if any:
         if (!performPullValidation(root)) {
             _pendingBlobs.clear();
-            _blob = _pendingBlobs.end();
+            _danglingBlobBegin = _blob = _pendingBlobs.end();
             return;
         }
 
-        // Request the first blob, or if there are none, insert the revision into the DB:
-        if (!_pendingBlobs.empty()) {
-            fetchNextBlob();
-        } else {
-            insertRevision();
+        _danglingBlobBegin = _pendingBlobs.end();
+        if (attachmentsFromSG.has_value()) {
+            vector<PendingBlob> dangled;
+            for (auto iter = _pendingBlobs.begin(); iter != _pendingBlobs.end();) {
+                auto digest = iter->key.digestString();
+                if (attachmentsFromSG->find(digest) == attachmentsFromSG->end()) {
+                    dangled.push_back(*iter);
+                    iter = _pendingBlobs.erase(iter);
+                } else {
+                    ++iter;
+                }
+            }
+            for (const auto& b: dangled) {
+                _pendingBlobs.push_back(b);
+            }
+            _blob = _pendingBlobs.begin();
+            _danglingBlobBegin = _blob + (_pendingBlobs.size() - dangled.size());
         }
+
```

Comment:
Same as my comment on the previous PR that I think we don't need to fetch any blobs if we found a dangling blob. Then the logic to move / append the dangling blobs to the end of pending blobs will not be needed as well.

Reply:
yes, this can simplified. I can simply not call startNextBlob().
Should I do it?

Reply:
I think it makes sense and did it.